src/nmodl/solve.cpp

```diff
@@ -126,20 +128,13 @@ void solvhandler() {
         lq = lq->next;
         method = SYM(lq);
         cvodemethod_ = 0;
-        if (method && strcmp(method->name, "after_cvode") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 1;
-        }
-        if (method && strcmp(method->name, "cvode_t") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 2;
-        }
-        if (method && strcmp(method->name, "cvode_v") == 0) {
-            method = (Symbol*) 0;
-            lq->element.sym = (Symbol*) 0;
-            cvodemethod_ = 3;
+        const std::unordered_map<std::string, int> methods = {{"after_cvode", 1},
```

Comment:
With such a very low number of items maybe `underered_map` is overkill. What do you think of a plain `std::vector<std::pair<std::string, int>>` and `std::find`?

Reply:
Done, though I will admit that now the code is IMHO less readable since the `unordered_map` API is much nicer than messing with `pair` in a `vector`.

Reply:
`std::map` is much more readable! See comment below.

Reply:
Perhaps it's an aside, but in my experience using containers will increase the code significantly compared to current string compairisons.  On every function invocation, the container will likely be rebuilt.

For instance, in this godbolt, I copied the vector impl: https://godbolt.org/z/1naM8oW1o
Compared to a `string` compare, it's nearly 6 times longer, ASM wise.

Reply:
I guess my question in that case is: is this part of the code performance-critical? If so, I'm fine with leaving the current implementation (still would like to fix that typo though).

As a(nother) side note: the `unordered_map` version is only ~2x longer than the current one (only w/GCC, w/clang the difference is more like 5-6x, though still shorter than `vector` or `map`): https://godbolt.org/z/4a8Peo3Wb

Reply:
Maybe make it `static const` to avoid reconstructing the map? I don't think that ASM size really matters here - but I would also move the map very close to the enum definition to lessen the chance of divergence.